semantics/src/namespace/scopes.rs

```diff
@@ -52,6 +54,7 @@ pub struct ContractScope {
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct FunctionScope {
+    pub span: Span,
```

Comment:
I tried to put the `FunctionAttributes` here but there's a cyclic dependency so the best I could have come up with would have been `Option<FunctionAttributes>` which leads to more verbose code down the road. So, I decided to put the `Span` here because that's all we need to lookup the `FunctionAttributes` on the `Context` (After a small refactoring of the context https://github.com/ethereum/fe/pull/105/commits/1c8a21721127f428dfe26f23b92fb1d8578f7fe3)